meterfs/meterfs.c

```diff
@@ -19,34 +19,263 @@
 #include <sys/mount.h>
 #include <string.h>
 #include <stddef.h>
+#include <sys/reboot.h>
 
 #include "meterfs.h"
 #include "files.h"
 #include "node.h"
 
-#define TOTAL_SIZE(f)           (((f)->filesz / (f)->recordsz) * ((f)->recordsz + sizeof(entry_t)))
-#define SECTORS(f, sectorsz)    (((TOTAL_SIZE(f) + sectorsz - 1) / sectorsz) + 1)
+#define TOTAL_SIZE(f)        (((f)->filesz / (f)->recordsz) * ((f)->recordsz + sizeof(entry_t)))
+#define SECTORS(f, sectorsz) (((TOTAL_SIZE(f) + (sectorsz) - 1) / (sectorsz)) + 1)
 
-#define LOG_INFO(str, ...)      do { if(0) printf(str "\n", ##__VA_ARGS__); } while(0)
+#define LOG_INFO(str, ...) do { if(0) printf(str "\n", ##__VA_ARGS__); } while(0)
+#define LOG_DEBUG(str, ...) do { if(0) printf(str "\n", ##__VA_ARGS__); } while(0)
 
+#define MIN_PARTITIONS_SECTORS_NB 8
+#define HEADER_SIZE               (HGRAIN + MAX_FILE_CNT * HGRAIN)
+#define JOURNAL_OFFSET            (2 * HEADER_SIZE)
+#define SPARE_SECTOR_OFFSET(ssz)  (JOURNAL_OFFSET + (ssz))
 
-#define MIN_PARTITIONS_SECTORS_NB    3
+#define UNRELIABLE_WRITE 0 /* DEBUG ONLY! */
+#if UNRELIABLE_WRITE
+static int unrwrcnt = 0;
+static int unrwrreboot = 0;
+#define UNRELIABLE_WRITE_TRIGGER        32
+#define UNRELIABLE_WRITE_REBOOT         1
+#define UNRELIABLE_WRITE_REBOOT_TRIGGER 5
+#warning "meterfs UNRELIABLE_WRITE is ON. Did you really intend that?"
+#endif
 
-static const unsigned char magic[4] = { 0xaa, 0x41, 0x4b, 0x55 };
+static const unsigned char magicConst[4] = { 0x66, 0x41, 0x4b, 0xbb };
+
+struct partialEraseJournal_s {
+	size_t sector;
+	size_t eraseOffset;
+	unsigned char magic[4];
+};
```

Comment:
It's ok, because this is de facto not a external data - it is always written and read by the same process on the same target. But packing struct a good idea - padding/aligment (not really in this case, but...) can change unexpectedly for some reason

Reply:
please note that we've recently changed `size_t` size for some targets in `libphoenix`, so You should not treat it as a fixed-size type (that time for riscv64 only, but still...) :P

Reply:
Well, it's not that essential like in other places but still can be source of issue in case of stooping flashsrv during write if upgrade occurs

Reply:
I've removed size_t from this struct